core/http-auth-aws/src/main/java/software/amazon/awssdk/http/auth/aws/internal/signer/SignerLoader.java

```diff
@@ -38,11 +37,7 @@ private SignerLoader() {
     }
 
     public static AwsV4aHttpSigner getSigV4aSigner() {
-        return get(AwsV4aHttpSigner.class, "software.amazon.awssdk.http.auth.aws.crt.AwsCrtV4aHttpSigner");
-    }
-
-    public static AwsS3V4HttpSigner getS3SigV4aSigner() {
-        return get(AwsS3V4HttpSigner.class, "software.amazon.awssdk.http.auth.aws.crt.AwsCrtS3V4aHttpSigner");
+        return get(AwsV4aHttpSigner.class, "software.amazon.awssdk.http.auth.aws.crt.internal.signer.DefaultAwsCrtV4aHttpSigner");
```

Comment:
> so a question for others in the Java team - is it ok for an depending on @SdkInternalApi like this?

I think it is, as far as I understand other modules can (and do) use internal APIs, the idea is that they don't directly use the classes but indirectly through other classes, and I think that's what we are doing here.

Reply:
Or keep AwsCrtV4aHttpSigner as SdkPublicApi interface, and have `create()` that returns `DefaultAwsCrtV4aHttpSigner` which implements `AwsCrtV4aHttpSigner` (not AwsV4aHttpSigner)

Reply:
> keep AwsCrtV4aHttpSigner as SdkPublicApi interface, and have create() that returns DefaultAwsCrtV4aHttpSigner which implements AwsCrtV4aHttpSigner (not AwsV4aHttpSigner).

I'm not sure I'm following to be honest, why do you think this is better?

Reply:
See similar discussion for event streaming implementation's dynamic loading - https://github.com/aws/aws-sdk-java-v2/pull/4270#discussion_r1303306489

Reply:
> Or I think this is better - keep AwsCrtV4aHttpSigner as SdkPublicApi interface, and have `create()` that returns `DefaultAwsCrtV4aHttpSigner` which implements `AwsCrtV4aHttpSigner` (not AwsV4aHttpSigner).

Because, then the dependency is on an interface, instead of a class, allowing us to replace the implementation if we needed to.

Also, it is similar to the existing pattern of the old Signer interfaces and Crt implementations.

Whether the interface is SdkPublicApi or SdkProtected, I can be convinced either way.

Reply:
Actually, saying `SdkPublicApi` on the interface, would mean we allow someone, maybe outside of SDK client, to take a direct dependency on the Crt V4a Signer, which I think is a reasonable use case to support.

Reply:
I made the `SignerLoader` and `DefaultAwsCrtV4aHttpSigner` are now annotated with `@SdkProtectedApi`. The idea being, even if the specific classes won't be used directly a backwards compatibility break will still break another module instead of just being contained within the same module.